server/client.go

```diff
@@ -202,11 +202,13 @@ type outbound struct {
 	sws int32         // Number of short writes, used for dynamic resizing.
 	pb  int32         // Total pending/queued bytes.
 	pm  int32         // Total pending/queued messages.
-	sg  *sync.Cond    // Flusher conditional for signaling.
+	sg  *sync.Cond    // Flusher conditional for signaling to writeLoop.
 	wdl time.Duration // Snapshot of write deadline.
-	mp  int32         // Snapshot of max pending.
-	fsp int32         // Flush signals that are pending from readLoop's pcd.
-	lft time.Duration // Last flush time.
+	mp  int32         // Snapshot of max pending for client.
+	fsp int32         // Flush signals that are pending per producer from readLoop's pcd.
+	lft time.Duration // Last flush time for Write.
+	lwb int32         // Last byte size of Write.
```

Comment:
Do we really need this since it seems that you are using it only in flushOutbound(), no?

Reply:
Yes but I have plans for this already for rate calculations, so prefer to leave in..